components/browser/session/src/main/java/mozilla/components/browser/session/storage/SessionStorage.kt

```diff
@@ -4,28 +4,39 @@
 
 package mozilla.components.browser.session.storage
 
+import mozilla.components.browser.session.Session
 import mozilla.components.browser.session.SessionManager
+import mozilla.components.concept.engine.Engine
+import mozilla.components.concept.engine.EngineSession
+
+data class SessionWithState(
+    val session: Session,
+    val selected: Boolean = false,
+    val engineSession: EngineSession? = null
+)
+typealias SessionsSnapshot = List<SessionWithState>
 
 /**
  * Storage component for browser and engine sessions.
  */
 interface SessionStorage {
 
     /**
-     * Persists the session state of the provided [SessionManager].
+     * Persists the provided [SessionsSnapshot] for a given [Engine].
```

Comment:
Would be nice to add that snapshot can be retrieved using [SessionManager.createSnapshot]...